lib/Sema/TypeCheckSwitchStmt.cpp

```diff
@@ -1075,19 +1087,19 @@ namespace {
           SmallVector<Space, 4> spaces;
           Space::decomposeDisjuncts(DC, uncovered.getType(), {}, spaces);
           diagnoseMissingCases(RequiresDefault::No, Space::forDisjunct(spaces),
-                               unknownCase);
-        } else {
+                               unknownCase, defaultCase);
+        } else if (!defaultCase) {
           diagnoseMissingCases(Switch->getCases().empty()
                                  ? RequiresDefault::EmptySwitchBody
                                  : RequiresDefault::UncoveredSwitch,
-                               uncovered, unknownCase);
+                               uncovered, unknownCase, defaultCase);
         }
         return;
```

Comment:
Maybe the comment in 1082-1084 should change to say what happens when the exhaustive switch is requested. It looks like the comment is saying that the code cannot reach here with a `default:` case, but now it can, and it seems that it is simply accepted (which was the behaviour before, IIUC).